engine/language-server/src/main/scala/org/enso/languageserver/http/server/BinaryWebSocketServer.scala

```diff
@@ -0,0 +1,174 @@
+package org.enso.languageserver.http.server
+
+import akka.NotUsed
+import akka.actor.{ActorRef, ActorSystem}
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model.{RemoteAddress, StatusCodes}
+import akka.http.scaladsl.model.ws.{BinaryMessage, Message, TextMessage}
+import akka.http.scaladsl.server.Directives._
+import akka.http.scaladsl.server.Route
+import akka.stream.scaladsl.{Flow, Sink, Source}
+import akka.stream.{CompletionStrategy, Materializer, OverflowStrategy}
+import akka.util.ByteString
+import com.typesafe.scalalogging.LazyLogging
+import org.enso.languageserver.http.server.BinaryWebSocketServer.Config
+import org.enso.languageserver.http.server.WebSocketControlProtocol.{
+  CloseConnection,
+  ConnectionClosed,
+  ConnectionFailed,
+  OutboundStreamEstablished
+}
+import org.enso.languageserver.util.binary.{
+  BinaryDecoder,
+  BinaryEncoder,
+  DecodingFailure
+}
+
+import scala.concurrent.duration._
+import scala.concurrent.{ExecutionContext, Future}
+
+/**
+  * A web socket server realizing binary protocol.
```

Comment:
This doesn't make sense in english.

```suggestion
  * A web socket server using a binary protocol.
```